src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/service/LocationService.kt

```diff
@@ -56,11 +52,49 @@ class LocationService(
       mapOf(
         "id" to location.id.toString(),
         "prisonId" to location.prisonId,
-        "code" to location.code,
+        "path" to location.pathHierarchy,
       ),
       null,
     )
 
     return location
   }
+
+  @Transactional
+  fun updateLocation(id: UUID, updateLocationRequest: UpdateLocationRequest): LocationDTO {
+    val locationToUpdate = locationRepository.findById(id)
+      .orElseThrow { LocationNotFoundException(id.toString()) }
+
+    updateLocationRequest.parentId?.let {
+      locationToUpdate.setParent(locationRepository.findById(it).getOrNull() ?: throw LocationNotFoundException(it.toString()))
+    }
+
+    locationToUpdate.setCode(updateLocationRequest.code)
+    locationToUpdate.locationType = updateLocationRequest.locationType
+    locationToUpdate.updatedBy = authenticationFacade.getUserOrSystemInContext()
+    locationToUpdate.whenUpdated = LocalDateTime.now(clock)
+
+    log.info("Updated Location [$id]")
+    telemetryClient.trackEvent(
+      "Updated Location",
+      mapOf(
+        "id" to id.toString(),
+        "prisonId" to locationToUpdate.prisonId,
+        "path" to locationToUpdate.getPathHierarchy(),
+      ),
+      null,
+    )
+
+    return locationToUpdate.toDto()
+  }
+
+  @Transactional
+  fun deleteLocation(id: UUID): LocationDTO {
```

Comment:
delete is interesting, since deleting in NOMIS can cause FK error - I wonder how we will deal with this